plugins/org.locationtech.udig.project.ui/src/org/locationtech/udig/project/ui/ApplicationGIS.java

```diff
@@ -370,8 +370,20 @@ public static IToolManager getToolManager() {
             }
         }
         return toolManager;
-   }
+    }
 
+    
+    /**
+     * Set the ToolManager singleton.
+     *
+     * @param toolManager the new tool manager
+     */
+    public static void setToolManager(IToolManager toolManager) {
```

Comment:
Where is this Method called from and what is the reason to (re)set toolManager Singleton?